src/Skia/Avalonia.Skia/WriteableBitmapImpl.cs

```diff
@@ -31,44 +33,58 @@ public WriteableBitmapImpl(PixelSize size, Vector dpi, PixelFormat? format = nul
             Dpi = dpi;
 
             var colorType = PixelFormatHelper.ResolveColorType(format);
-            
+
             var runtimePlatform = AvaloniaLocator.Current?.GetService<IRuntimePlatform>();
-            
+
             if (runtimePlatform != null)
             {
+                var nfo = new SKImageInfo(size.Width, size.Height, colorType, SKAlphaType.Premul);
+
+                _blob = runtimePlatform.AllocBlob(nfo.BytesSize);
+
+                _pixMap = new SKPixmap(nfo, _blob.Address, nfo.RowBytes);
+
                 _bitmap = new SKBitmap();
 
-                var nfo = new SKImageInfo(size.Width, size.Height, colorType, SKAlphaType.Premul);
-                var blob = runtimePlatform.AllocBlob(nfo.BytesSize);
+                _bitmap.InstallPixels(_pixMap);
+
+                _bitmap.SetImmutable();
 
-                _bitmap.InstallPixels(nfo, blob.Address, nfo.RowBytes, s_releaseDelegate, blob);
+                _image = SKImage.FromBitmap(_bitmap);
```

Comment:
Why do you need SKImage here? GPU bitmap caching works for both SKImage and SKBitmap. Creating SKImage doesn't magically move pixels to GPU memory.

Reply:
I think it is better to use SKImage everywhere to get the same results for all images.

Reply:
With this change, the `IDrawableBitmapImpl` will be redundant.

Reply:
It won't be redundant since some bitmaps (e. g. a user GL texture wrapped in a *temporary* SKImage) will need custom drawing and lifetime management logic

Reply:
Isn't the `OptionalDispose<T>` approach the better solution for that? But we shouldn't change anything just for the change you are right. Having all drawing code in one place just feels better.

Reply:
A "bitmap" might also be a vector `SKPicture`, which can't be properly represented by SKImage (SKImage::FromPicture makes a rasterized copy).

Reply:
But in the end it will be a SkImage. So it would work with the way the bitmaps are implemented for Direct2D1 backend.

Reply:
Why do we need to allocate a blob here instead of letting Skia take care of it?

Reply:
Blobs are easier to track and analyze in the memory profiler.